src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -524,6 +524,18 @@ private void AddFunctionDescriptors()
             {
                 _logger.AddingDescriptorProviderForLanguage(_workerRuntime);
                 _descriptorProviders.Add(new RpcFunctionDescriptorProvider(this, _workerRuntime, ScriptOptions, _bindingProviders, _functionDispatcher, _loggerFactory, _applicationLifetime));
+                AddCodelessDescriptor();
+            }
+        }
+
+        /// <summary>
+        /// Adds a DotNetFunctionDescriptorProvider to the list of descriptors if any function metadata has language set to "codeless" in it.
+        /// </summary>
+        private void AddCodelessDescriptor()
+        {
+            if (_functionMetadataManager.Functions.Any(m => string.Equals(m?.Language, DotNetScriptTypes.Codeless, StringComparison.OrdinalIgnoreCase)))
```

Comment:
We should remove the special handling in this file for IProxyMetadataManager and treat it the same as these custom providers. If we register ProxyMetadataManager as an IFunctionMetadataProvider with DI that should happen naturally, and we can register ProxyFunctionDescriptorProvider in AddFunctionDescriptors as well, where it should be. Not sure why we're doing that in GetFunctionsMetadata right now